### ISSUE TITLE
Inferred type of variable is partially unknown within loop

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#2362]
This is related to an earlier bug https://github.com/microsoft/pyright/issues/2288.

```python
def build_namespaces(namespace: str) -> List[str]:
    parts = namespace.split('.')
    namespaces: List[str] = []
    ns = ''
    for part in parts:
        if ns:
            ns += f'.{part}'
        else:
            ns += part
        namespaces.append(ns)
    return namespaces
```

**Comment by erictraut** [comment]
Here's a simplified repro:

```python
# pyright: strict

def func(x: list[str]):
    n = ""
    while True:
        if n:
            n = n + ""
        x.append(n)
```

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.174, which I just published. It will also be included in the next release of pylance.
